index.js

```diff
@@ -32,3 +32,22 @@ module.exports = fn => {
 		};
 	};
 };
+
+module.exports.proxy = fn => {
+	const cache = [];
```

Comment:
I would also use `Reflect.get()` and `Reflect.apply()` here for safety and clarity of intent.